server/belga/search_providers.py

```diff
@@ -297,9 +300,30 @@ def find(self, query, params=None):
             api_params['searchText'] = ''
 
         data = self.api_get(self.search_endpoint, api_params)
-        docs = [self.format_list_item(item) for item in data[self.items_field]]
+        docs = [self.get_detailed_info(item) for item in data[self.items_field]]
```

Comment:
I'm not sure if we can do this performance wise, doing extra request for each item means quite some load. imo would be better to only fetch the detail for item preview, that might require some core changes

Reply:
yep, I was thinking the same, we have to do changes in core probably so I will check it.